reproman/utils.py

```diff
@@ -1383,4 +1384,33 @@ def merge_dicts(ds):
     return merged
 
 
+def pycache_source(path):
+    """Map a pycache path to the original path.
+
+    Parameters
+    ----------
+    path : str
+        A Python cache file.
+
+    Returns
+    -------
+    Path of cached Python file (str) or None if `path` doesn't look like a
+    cache file.
+    """
+    if not path.endswith(".pyc"):
```

Comment:
so we will not bother with `.pyo`  (from `python -O`) since they are unlikely to encounter?

If you decide to add support for them, note that with py3 they seems to not acquire `.pyo` extension like with python2 but rather get additional suffix to the filename.  Here is a result of little local exploration of importing a module with and without `-O`
```shell
$> find hello 
hello
hello/__init__.pyc
hello/__init__.pyo
hello/__pycache__
hello/__pycache__/__init__.cpython-37.opt-1.pyc
hello/__pycache__/__init__.cpython-37.pyc
hello/__init__.py
```
I ran `locate` on my laptop for .opt-1.pyc and found none, but for .pyo did find quite a few hits in virtual environments around the projects where I am not sure if I ever ran them explicitly with `python -O`, e.g. even in our `simple_workflow`
```
/home/yoh/proj/repronim/reprozip/reprounzip-docker/reprounzip/__init__.pyo
/home/yoh/proj/repronim/reprozip/reprounzip-docker/reprounzip/unpackers/__init__.pyo
/home/yoh/proj/repronim/simple_workflow/simple_workflow/miniconda/envs/bh_demo/lib/python2.7/BaseHTTPServer.pyo
/home/yoh/proj/repronim/simple_workflow/simple_workflow/miniconda/envs/bh_demo/lib/python2.7/Bastion.pyo
/home/yoh/proj/repronim/simple_workflow/simple_workflow/miniconda/envs/bh_demo/lib/python2.7/CGIHTTPServer.pyo
/home/yoh/proj/repronim/simple_workflow/simple_workflow/miniconda/envs/bh_demo/lib/python2.7/ConfigParser.pyo
/home/yoh/proj/repronim/simple_workflow/simple_workflow/miniconda/envs/bh_demo/lib/python2.7/Cookie.pyo
/home/yoh/proj/repronim/simple_workflow/simple_workflow/miniconda/envs/bh_demo/lib/python2.7/DocXMLRPCServer.pyo
```
so may be after all, while at it, worthwhile adding support for those?

Reply:
> so we will not bother with `.pyo` (from `python -O`) since they are unlikely to encounter?

No, I just didn't think about it.  Thanks for bringing it up.